HandEyeCalibration/UtilizeEyeInHandCalibration/UtilizeEyeInHandCalibration.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+Utilize the result of eye-in-hand calibration to transform (picking) point
+coordinates from the camera frame to the robot base frame.
+*/
+
+#include <Eigen/Core>
+#include <opencv2/core/core.hpp>
+
+#include <cmath>
+#include <iostream>
+
+Eigen::MatrixXd cvToEigen(const cv::Mat &cvMat)
+{
+    if(cvMat.dims > 2)
+    {
+        throw std::exception("Invalid matrix dimensions. Expected 2D.");
+    }
+
+    Eigen::MatrixXd eigenMat(cvMat.rows, cvMat.cols);
+
+    for(int i = 0; i < cvMat.rows; i++)
+    {
+        for(int j = 0; j < cvMat.cols; j++)
+        {
+            eigenMat(i, j) = cvMat.at<double>(i, j);
+        }
+    }
+
+    return eigenMat;
+}
+
+cv::Mat readTransform(const std::string &file_name)
+{
```

Comment:
I think we can make some more improvements to error handling in this function.

Could `fileStorage["PoseState"]` or `poseStateNode.mat()` throw? If so, the file storage still won't be released. And if someone comes along in the future and adds more potentially throwing code in this function, they might forget to release the file. I think it would be better to wrap anything that might throw inside a `try` block and release the file in the `catch`, then we're protected against these cases as well. (Some sort of scope guard might be even cleaner, but I suggest using try/catch instead to make samples as simple as possible.)

The exception for pose state not found and invalid dimensions is a bit ambiguous. It would be better to do one check at a time and throw a more specific error.

There seems to be several compilation errors in this function now.

Here's a suggestion for fixing all of the above. It's not been tested or formatted correctly, but hopefully it will be useful to you as a starting point:

```c++
cv::Mat readTransform(const std::string &file_name)
{
    auto fileStorage = cv::FileStorage();

    if(!fileStorage.open(file_name, cv::FileStorage::Mode::READ))
    {
        throw std::invalid_argument("Could not open " + file_name);
    }
    try
    {
        const auto poseStateNode = fileStorage["PoseState"];

        if(poseStateNode.empty())
        {
            throw std::invalid_argument("PoseState not found in file " + file_name);
        }

        const auto rows = poseStateNode.mat().rows;
        const auto cols = poseStateNode.mat().cols;
        if(rows != 4 || cols != 4)
        {
            throw std::invalid_argument("Expected 4x4 matrix in " + file_name + ", but got " + std::to_string(cols) + "x" + std::to_string(rows));
        }

        const auto poseState = poseStateNode.mat();
        fileStorage.release();
        return poseState;
    }
    catch(...)
    {
        fileStorage.release();
        throw;
    }
}
```